tremor-pipeline/src/lib.rs

```diff
@@ -72,6 +76,45 @@ pub use op::{ConfigImpl, InitializableOperator, Operator};
 pub use tremor_script::prelude::EventOriginUri;
 pub(crate) type ExecPortIndexMap =
     HashMap<(usize, Cow<'static, str>), Vec<(usize, Cow<'static, str>)>>;
+#[derive(Debug)]
+///Pluggable-logging Appender
+pub struct PluggableLoggingAppender {
+    /// async Sender
+    pub tx: Sender<LoggingMsg>,
+}
+
+impl Append for PluggableLoggingAppender {
+    fn append(&self, record: &log::Record) -> anyhow::Result<()> {
+        let vec = (r#"{"level": ""#.to_owned()
```

Comment:
I think it would be better to use the `literal!()` macro here, so we get a `Value`, instead of creating a string.
This way can cause problems, eg. when the line contains `"`.
Also, please remove the `expect`, as Tremor should never panic in production, it can be replaced e.g. with `unwrap_or`